utils/terrascript_client.py

```diff
@@ -306,6 +310,66 @@ def populate_additional_providers(self, accounts):
                            alias=alias,
                            assume_role={'role_arn': assume_role})
 
+    def populate_route53(self, desired_state):
+        for zone in desired_state:
+            # provider = zone['provider']
+            account = zone['account']
+            account_name = account['name']
+            origin = zone['origin']
+            records = zone['records']
+
+            # terraform identifier for zone
+            # We need to replace some characters that are not allowed
+            zone_id = origin
+            zone_id = zone_id.replace(".", "__")
+
+            values = {
+                'name': origin,
+            }
+
+            zone_resource = aws_route53_zone(zone_id, **values)
+            self.add_resource(account_name, zone_resource)
+
+            counts = defaultdict(int)
+            for record in records:
+                counts[record['name']] += 1
+
+                # Terraform record id
+                # This contains a lot of info on purpose
+                # - Needs to be unique per record per zone
+                # - Adding a count for when we have record with the same name
+                # - Terraform client logs only the resource name, so having
+                #   all this info as actually useful when looking at reports
+                record_id = record['name'] + "_" + \
+                    str(counts[record['name']]) + "__" + \
+                    zone_id + "__" + \
+                    record['type'] + "__" + \
+                    record['target'].replace(".", "__")
+
+                # terraform identifier for record
+                # We need to replace some characters that are not allowed
+                record_id = record_id.replace(".", "__")
+                record_id = record_id.replace("@", "_ORIGIN_")
+                record_id = record_id.replace("*", "_wildcard_")
+
+                values = {
+                    'zone_id': zone_resource.fullname,
+                    'name': record['name'],
```

Comment:
I'd reduce the number of dict lookups for the same key by placing it in a variable on the first lookup. Not a big difference thoug, just fine tunning.

Reply:
Easy optimisation. Done!